website/server/controllers/top-level/dataexport.js

```diff
@@ -243,4 +243,49 @@ api.exportUserAvatarPng = {
   },
 };
 
+/**
+ * @api {get} /export/inbox.html Export user private messages as HTML document
+ * @apiVersion 3.0.0
+ * @apiName ExportUserPrivateMessages
+ * @apiGroup DataExport
+ * @apiDescription NOTE: Part of the private API that may change at any time.
+ *
+ * @apiSuccess {String} An html page
+ */
+api.exportUserPrivateMessages = {
+  method: 'GET',
+  url: '/export/inbox.html',
+  middlewares: [authWithSession],
+  async handler (req, res) {
+    const timezoneOffset = res.locals.user.preferences.timezoneOffset;
+    const dateFormat = res.locals.user.preferences.dateFormat.toUpperCase();
+
+    let inbox = Object.keys(res.locals.user.inbox.messages).map(key => res.locals.user.inbox.messages[key]);
+
+    // Sort array in descending order
+    inbox = _.sortBy(inbox, function sortBy (num) {
+      return num.sort * -1;
+    });
+
+    let messages = '<!DOCTYPE html><html><head></head><body>';
+
+    let recipient;
+
+    for (let i = 0; i < inbox.length; i++) {
+      recipient = inbox[i].sent ? '=>' : '<=';
+      messages += `<p>${recipient}<strong>${inbox[i].user}</strong> ${moment.utc(inbox[i].timestamp).zone(timezoneOffset).format(`${dateFormat } HH:mm:ss`)} 
```

Comment:
This is kind of messy looking. Can we pull some of the logic out into variables and do some line breaks?

```js
let recipientIcon = inbox[i].sent ? '=>' : '<=';
let { user, text } = inbox[i];
let timestamp = moment.utc(inbox[i].timestamp).zone(timezoneOffset).format(`${dateFormat } HH:mm:ss`);
let index = `(${i + 1}/${inbox.length});
messages = `
<p>
  ${recipientIcon}<strong>${user}</strong> ${timestamp}
  ${index}
  <br />
  ${text}
</p>`;
```